internal/hcp/registry/types.bucket.go

```diff
@@ -638,12 +642,19 @@ func (bucket *Bucket) completeBuild(
 		}
 
 		state := art.State(packerSDKRegistry.ArtifactStateURI)
+		if state == nil {
+			return packerSDKArtifacts, &NotAHCPArtifactError{
```

Comment:
I would expect the returned artifacts to be nil 
```suggestion
			return nil, &NotAHCPArtifactError{
```

Reply:
Given that the caller keeps the artifacts returned over the ones passed as argument (with the added HCP registry artifact in case of success) I'm not certain returning `nil` here is a good idea as it would make reporting the artifact to stdout after the build impossible.
Honestly this function could change to return only the artifact to be added so we don't override the argument with `nil` if it fails for some reason, but this could be done in a later PR

Reply:
Oh okay - that makes sense. There is magic happening for sure. Sorry I'm forgetting how this is all wired.